kw6/stream.py

```diff
@@ -45,7 +45,37 @@ def from_path(path, header_path=None):
             cached_positions=header_positions(header_path) if header_path is not None else {},
         )
 
-    def seek_(self, frame_index: types.FRAME_INDEX):
+    def __getitem__(self, indices_or_slice):
+        if type(indices_or_slice) == int:
+            if indices_or_slice < 0:
+                raise IndexError('Negative indexing not supported.')
+            self._seek(indices_or_slice)
+            if not self.empty():
+                positions = Position.from_stream_(self.stream)
+            else:
+                raise IndexError(indices_or_slice)
+
+        elif type(indices_or_slice) == slice:
```

Comment:
Maybe it's better to skip slicing? We are not supporting slices fully here since they can also use step size?

Reply:
Can you actually index an array with a step size? I didn't think you could and that's why I didn't implement it, it could certainly be added

Reply:
Okay sure, you can add an exception if you get a stepsize != 1 I guess?

Reply:
> Okay sure, you can add an exception if you get a stepsize != 1 I guess?

Done